sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -153,15 +157,113 @@ private[spark] class HiveExternalCatalog(client: HiveClient, hadoopConf: Configu
     requireDbExists(db)
     requireDbMatches(db, tableDefinition)
 
-    if (
+    if (tableDefinition.provider == Some("hive") ||
+        tableDefinition.tableType == CatalogTableType.VIEW) {
+      client.createTable(tableDefinition, ignoreIfExists)
+    } else {
+      import CreateDataSourceTableUtils._
+
+      val provider = tableDefinition.provider.get
+      val userSpecifiedSchema = tableDefinition.userSpecifiedSchema
+      val partitionColumns = tableDefinition.partitionColumnNames
+
+      val tableProperties = new mutable.HashMap[String, String]
+      tableProperties.put(DATASOURCE_PROVIDER, provider)
+
+      // Saves optional user specified schema.  Serialized JSON schema string may be too long to be
+      // stored into a single metastore SerDe property.  In this case, we split the JSON string and
+      // store each part as a separate SerDe property.
+      userSpecifiedSchema.foreach { schema =>
+        val schemaJsonString = schema.json
+        // Split the JSON string.
+        val parts = schemaJsonString.grouped(4000).toSeq
```

Comment:
the `schemaStringLengthThreshold` is dropped as we can't access sql conf here, is it OK?